src/services/codefixes/fixAddMissingMember.ts

```diff
@@ -405,4 +429,87 @@ namespace ts.codefix {
+            return enumMember === undefined || name === undefined ? factory.createNumericLiteral(0) : factory.createPropertyAccessExpression(name, checker.symbolToString(enumMember));
+        }
+        if (type.flags & TypeFlags.NumberLiteral) {
+            return factory.createNumericLiteral((type as NumberLiteralType).value);
+        }
+        if (type.flags & TypeFlags.BigIntLiteral) {
+            return factory.createBigIntLiteral((type as BigIntLiteralType).value);
+        }
+        if (type.flags & TypeFlags.StringLiteral) {
+            return factory.createStringLiteral((type as StringLiteralType).value, /* isSingleQuote */ quotePreference === QuotePreference.Single);
+        }
+        if (type.flags & TypeFlags.BooleanLiteral) {
+            return (type === checker.getFalseType() || type === checker.getFalseType(/*fresh*/ true)) ? factory.createFalse() : factory.createTrue();
+        }
+        if (type.flags & TypeFlags.Null) {
+            return factory.createNull();
+        }
+        if (type.flags & TypeFlags.Union) {
+            const expression = firstDefined((type as UnionType).types, t => tryGetInitializerValueFromType(context, checker, importAdder, quotePreference, t));
+            return expression ?? createUndefined();
+        }
+        if (checker.isArrayLikeType(type)) {
+            return factory.createArrayLiteralExpression();
+        }
+        if (getObjectFlags(type) & ObjectFlags.Anonymous) {
+            const decl = find(type.symbol.declarations || emptyArray, or(isFunctionTypeNode, isMethodSignature, isMethodDeclaration));
+            if (decl === undefined) return createUndefined();
+
+            const signature = checker.getSignaturesOfType(type, SignatureKind.Call);
+            if (signature === undefined) return createUndefined();
+
+            const func = createSignatureDeclarationFromSignature(SyntaxKind.FunctionExpression, context, quotePreference, signature[0],
+                createStubbedBody(Diagnostics.Function_not_implemented.message, quotePreference), /*name*/ undefined, /*modifiers*/ undefined, /*optional*/ undefined, /*enclosingDeclaration*/ undefined, importAdder) as FunctionExpression | undefined;
+            return func ?? createUndefined();
+        }
+        if (getObjectFlags(type) & ObjectFlags.Class) {
+            const classDeclaration = getClassLikeDeclarationOfSymbol(type.symbol);
+            if (classDeclaration === undefined || hasAbstractModifier(classDeclaration)) return createUndefined();
+
+            const constructorDeclaration = getFirstConstructorWithBody(classDeclaration);
+            if (constructorDeclaration && length(constructorDeclaration.parameters)) return createUndefined();
+
+            return factory.createNewExpression(factory.createIdentifier(type.symbol.name), /*typeArguments*/ undefined, /*argumentsArray*/ undefined);
+        }
+        return createUndefined();

```

Comment:
Need a case for `TypeFlags.TemplateLiteral`?

Reply:
@andrewbranch Thanks for the feedback. Could you help with an example to cover the case with the `TemplateLiteral`?

Reply:
```ts
interface I1 {
   cssVariable: `--${string}`;
}

const a: I1 = {};
```

I would probably just throw that into the `string` case and seed it with an empty string. No need to try to come up with a placeholder that actually fits the template 😄

Reply:
@andrewbranch Oke. Thanks. I've added a case for `TemplateLiteral ` and a test that covers the strings quote option.